bloom-housing/bloom
# refactor: move public specific components into public site folder

# Pull Request Template

## Issue Overview

This PR addresses #3238


## Description

This PR moves some components out of UIC and into the public site that are specific to only the public site. I have tabled `ApplicationTimeline` for now because it will be a bigger effort - the `ApplicationTimeline` component has a strange approach where it pulls in class names from the `ProgressNav` and overwrites them but doesn't do so with V2 styling (followup [issue](https://github.com/bloom-housing/bloom/issues/3259)).

It also reorganizes the public site directory - because we are adding more individual components, leaving it as-is was just a little confusing. The update is based on best practices from perusing articles. The directory changes are as follows:

<img width="188" alt="Screenshot 2023-01-11 at 1 57 36 PM" src="https://user-images.githubusercontent.com/65367387/211917851-287e1170-0873-4db2-a4b9-659d88457249.png">

* All site content now lives under the `/src` directory.
* There is a new `components` folder organized by each section of the application. This is where the shuffled components ended up. I would encourage us to move some of our contained UI pieces into this folder to encourage modularization, and it would make it a lot easier to test them (lookin at you `ListingView.tsx`).
* One of the components brought over had its own scss file. For file-level CSS, NextJs by default supports CSS modules. You can see this pattern with `StatusItem`. I would want to remove Tailwind and continue to lean on CSS modules to keep ourselves tidy with our own design system.

## How Can This Be Tested/Reviewed?

The only code change was in the `StatusItem` component, since it was moved to use CSS modules (this component renders on the my applications page within accounts and is a summary of a submitted application). I would ensure there are not visual regressions by submitting an application and checking out the component.

Otherwise, take a look at the directory structure and open to feedback about how to make things more organized / readable!

There are six test files from the components brought over in a new `__tests__` folder that should also be running successfully in a new job below (`unit-test-public`).

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
